Building Docker images on main branch fails

**Describe the bug**
Docker image build fails with the following error:

```
+ cd ballista/scheduler/ui
[2422](https://github.com/paolorechia/arrow-ballista/actions/runs/4517673072/jobs/7957036021#step:3:2423)
+ yarn install
[2423](https://github.com/paolorechia/arrow-ballista/actions/runs/4517673072/jobs/7957036021#step:3:2424)
yarn install v1.22.19
[2424](https://github.com/paolorechia/arrow-ballista/actions/runs/4517673072/jobs/7957036021#step:3:2425)
[1/4] Resolving packages...
[2425](https://github.com/paolorechia/arrow-ballista/actions/runs/4517673072/jobs/7957036021#step:3:2426)
[2/4] Fetching packages...
[2426](https://github.com/paolorechia/arrow-ballista/actions/runs/4517673072/jobs/7957036021#step:3:2427)
error sockjs-client@1.6.1: The engine "node" is incompatible with this module. Expected version ">=12". Got "10.24.0"
[2427](https://github.com/paolorechia/arrow-ballista/actions/runs/4517673072/jobs/7957036021#step:3:2428)
error Found incompatible module.
```

**To Reproduce**
```
git clone git@github.com:apache/arrow-ballista.git
cd arrow-ballista
./dev/build-ballista-docker.sh
```

**Expected behavior**
Ballista docker images are successfully built. 

**Additional context**
It seems like the node version used by the docker image is no longer supported by the ` sockjs-client@1.6.1`. It seems we'd have two options to solve this:

1. Downgrade ` sockjs-client@1.6.1`
2. Upgrade the base node version used in the build.

Perhaps one option going with 2 is to integrate nvm (Node Version Manager: https://github.com/nvm-sh/nvm#install--update-script) into the docker build, so we can easily update the used node version? I might give this a go if I have time.

Building still fails on main (commit 9b4a12191aac37bda6d0c5e933377698687872b9).

> Compiling pest_generator v2.5.6
> error[E0004]: non-exhaustive patterns: `pest_meta::optimizer::OptimizedExpr::NodeTag(_, _)` not covered
>    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/pest_generator-2.5.6/src/generator.rs:378:11
>     |
> 378 |     match expr {
>     |           ^^^^ pattern `pest_meta::optimizer::OptimizedExpr::NodeTag(_, _)` not covered
>     |
> note: `pest_meta::optimizer::OptimizedExpr` defined here
>    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/pest_meta-2.6.0/src/optimizer/mod.rs:142:5
>     |
> 114 | pub enum OptimizedExpr {
>     | ----------------------
> ...
> 142 |     NodeTag(Box<OptimizedExpr>, String),
>     |     ^^^^^^^ not covered
>     = note: the matched value is of type `pest_meta::optimizer::OptimizedExpr`
> help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
>     |
> 519 ~         }
> 520 +         pest_meta::optimizer::OptimizedExpr::NodeTag(_, _) => todo!()
>     |
> 
> error[E0004]: non-exhaustive patterns: `pest_meta::optimizer::OptimizedExpr::NodeTag(_, _)` not covered
>    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/pest_generator-2.5.6/src/generator.rs:524:11
>     |
> 524 |     match expr {
>     |           ^^^^ pattern `pest_meta::optimizer::OptimizedExpr::NodeTag(_, _)` not covered
>     |
> note: `pest_meta::optimizer::OptimizedExpr` defined here
>    --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/pest_meta-2.6.0/src/optimizer/mod.rs:142:5
>     |
> 114 | pub enum OptimizedExpr {
>     | ----------------------
> ...
> 142 |     NodeTag(Box<OptimizedExpr>, String),
>     |     ^^^^^^^ not covered
>     = note: the matched value is of type `pest_meta::optimizer::OptimizedExpr`
> help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
>     |
> 651 ~         }
> 652 +         pest_meta::optimizer::OptimizedExpr::NodeTag(_, _) => todo!()
>     |
> 
> For more information about this error, try `rustc --explain E0004`.
> error: could not compile `pest_generator` due to 2 previous errors
> warning: build failed, waiting for other jobs to finish..

--- a/ballista/scheduler/ui/package.json
+++ b/ballista/scheduler/ui/package.json
@@ -28,8 +28,8 @@
     "web-vitals": "^1.0.1"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
+    "start": "react-scripts --openssl-legacy-provider start",
+    "build": "react-scripts --openssl-legacy-provider build",
     "test": "react-scripts test",
     "eject": "react-scripts eject"
   },

--- a/dev/docker/ballista-builder.Dockerfile
+++ b/dev/docker/ballista-builder.Dockerfile
@@ -24,8 +24,11 @@ ENV RUST_BACKTRACE=full
 ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && \
-    apt-get -y install libssl-dev openssl zlib1g zlib1g-dev libpq-dev cmake protobuf-compiler netcat curl unzip \
-    nodejs npm && \
+    apt-get -y install libssl-dev openssl zlib1g zlib1g-dev libpq-dev cmake protobuf-compiler netcat curl unzip
+
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
+    apt-get update && \
+    apt-get install -y nodejs && \
     npm install -g yarn
 
 # create build user with same UID as 

